ndcube/utils/collection.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import astropy.units as u
+
+
+def _sanitize_aligned_axes(data, aligned_axes, n_cubes):
```

Comment:
The length of this function hurts my brain.

Without looking over it, I feel like if we need this much code to parse the input, our input format is too complex. Is there a way we can split this into two argument, or disallow some things (we could drop int and enforce a len 1 tuple etc).